### DISCUSSION TITLE
How do I close manually after the command has been successfully executed?

### DISCUSSION THREAD

**Discussion by SabineMueller910** [jmcdo29/nest-commander#1117] [Q&A] [answered]
I have written a command that fills the database with data. Unfortunately I had to use `CommandFactory.runWithoutClosing()` to make it work, otherwise `run()` will not work and the database connection will not be maintained while the command is running.
Is there any other way to ensure that the application is closed **after** the database command has finished?

I am grateful for any suggestions.

**Comment by jmcdo29** [comment, marked as answer]
Why do you need `runWithoutClosing` here? So long as everything in the command is properly awaited, the command shouldn't be "finished" until all of the queries are executed and the connection is closed

**Reply by SabineMueller910** [reply]
Unfortunately not. Everything is being waited for. I can't find any reason in my application for this problem. The error also occurs when I debug the code and run an `insert` or even a `select` command on the database. It stops while processing the operation with a message like this: `Trying to check out a connection from a closed connection pool`.

**Reply by jmcdo29** [reply]
Interesting. Can you provide a reproduction, because I don't think that should be the expected outcome

**Reply by SabineMueller910** [reply]
Thank you for drawing my attention to the problem once again and I have checked all the places in the code again to see whether the result is always waited for there.
I did indeed forget it in one place, precisely in the Commander app 🙈.


Now the application stops and no longer terminates with errors when I run it with "run()".

Thanks a lot!!!
